app/src/main/java/com/sdpteam/connectout/event/EventCreatorModel.java

```diff
@@ -0,0 +1,88 @@
+package com.sdpteam.connectout.event;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+public class EventCreatorModel implements EventDataManager {
+    public final static String DATABASE_EVENT_PATH = "Events";
+    private final DatabaseReference database;
+
+    public EventCreatorModel() {
+        database = FirebaseDatabase.getInstance().getReference();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public boolean saveValue(Event event) {
+        if (event != null) {
+            database.child(DATABASE_EVENT_PATH).child(event.getEventId()).setValue(event);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    @Override
+    public LiveData<Event> getValue(String eid) {
+        MutableLiveData<Event> value = new MutableLiveData<>();
+        database.child(DATABASE_EVENT_PATH).child(eid).addListenerForSingleValueEvent(new ValueEventListener() {
```

Comment:
For future readers, 
Please note that the below approach of using a combined listener `IdValueListener` for both models is chosen over a generic model since, the `DataReference` path resolution in both `ProfileModel` and `EventCreatorModel` would result in too much ambiguity.